src/t_hash.c

```diff
@@ -235,14 +238,16 @@ static uint64_t dictMstrHash(const void *key) {
 }
 
 static void dictHfieldDestructor(dict *d, void *field) {
+    size_t usable, *alloc_size = (size_t *)dictMetadata(d);
```

Comment:
maybe declare static function `htMetadata *getMetaPtr(dict *d)`
(See my next comment about  `htMetadata`)

Reply:
Done in [807dd7b](https://github.com/redis/redis/pull/14363/commits/807dd7b20d377e6a3c05dca0e57c027e0aafac8d)